pipeline_tools/input_utils.py

```diff
@@ -1,335 +1,168 @@
-    """Returns fastq file names
+        bundle (humancellatlas.data.metadata.Bundle): A Bundle object contains all of the necessary information.
 
-    Args:
-        assay_json (dict): metadata about the assay
-    
     Returns:
-        tuple of two strings representing the two fastq names
+        sample_id (str): String giving the sample id
     """
-    fastq_1_name = assay_json["content"]["seq"]["lanes"][0]["r1"]
-    fastq_2_name = assay_json["content"]["seq"]["lanes"][0]["r2"]
-    return fastq_1_name, fastq_2_name
+    cell_suspension = bundle.sequencing_input[0]
+    sample_id = str(cell_suspension.document_id)
+    return sample_id
 
 
-def get_optimus_lanes(metadata_json, version):
-    """Get the lane metadata
+def get_urls_to_files_for_ss2(bundle):
+    """Return the direct urls to the input fastq files for SmartSeq2 pipeline.
 
     Args:
-        metadata_json (dict): metadata
-        version (str): the metadata version
+        bundle (humancellatlas.data.metadata.Bundle): A Bundle object contains all of the necessary information.
 
     Returns:
-        Dict of lane metadata 
-
-    Raises:
-        NotImplementedError: if metadata version is unsupported
+         tuple: A tuple consisting of the url to the input fastq files, which are corresponding to read index 1 and 2
+         respectively.
     """
-    if version.startswith('4.'):
-        return _get_optimus_lanes_v4(metadata_json)
-    else:
-        raise NotImplementedError('Only implemented for v4 metadata')
+    fastq_1_url = fastq_2_url = None
+    sequence_files = bundle.sequencing_output
 
+    for sf in sequence_files:
+        if fastq_1_url and fastq_2_url:
+            return fastq_1_url, fastq_2_url  # early termination
+        if sf.read_index == 'read1':
+            fastq_1_url = sf.manifest_entry.url
+        if sf.read_index == 'read2':
+            fastq_2_url = sf.manifest_entry.url
+    return fastq_1_url, fastq_2_url
 
-def _get_optimus_lanes_v4(assay_json):
-    """Return the lane metadata from the assay json"""
-    lanes = assay_json['content']['seq']['lanes']
-    return lanes
 
+def download_file(item, dss_url, http_requests=HttpRequests()):
+    """Download the metadata for a given bundle from the HCA data store (DSS).
 
-def get_optimus_inputs(lanes, manifest_files):
-    """Returns metadata for optimus input files
+    This function borrows a lot of existing code from the `metadata-api` code base for consistency,
+    and this won't be required after migrating to use the HCA DSS Python API `dss_client` directly.
 
     Args:
-        lanes (dict): lane metadata
-        manifest_files (dict): file metadata
-
-    Returns:
-        Three lists of urls, representing fastqs for r1, r2, and i1, respectively.
-    In each list, the first item is for the first lane, the second item is for the second lane, etc.
+        item (typing.ItemsView): A dictionary's ItemsView object consisting of file_name and the manifest_entry.
```

Comment:
Is `manifest_entry` the same as the `manifest` json file? If so, let's use the same name for consistency

Reply:
Oh nevermind! I see that now that the `manifest_entry` is the info about a particular item in the `manifest` json file

Reply:
yep, before going to JSON, it's now a class defined in "metadata-api"